pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/UserGovernanceCachedDao.java

```diff
@@ -0,0 +1,79 @@
+package org.broadinstitute.ddp.db.dao;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.broadinstitute.ddp.cache.ModelChangeType;
+import org.broadinstitute.ddp.model.governance.Governance;
+import org.jdbi.v3.core.Handle;
+
+public class UserGovernanceCachedDao extends SQLObjectWrapper<UserGovernanceDao> implements UserGovernanceDao {
+    public UserGovernanceCachedDao(Handle handle) {
+        super(handle, UserGovernanceDao.class);
+    }
+
+    public Governance createGovernedUser(long clientId, long proxyUserId, String alias) {
+        var result = target.createGovernedUser(clientId, proxyUserId, alias);
+        notifyModelUpdated(ModelChangeType.USER, proxyUserId);
+        return result;
+    }
+
+    public Governance createGovernedUserWithGuidAlias(long clientId, long proxyUserId) {
+        var result = target.createGovernedUserWithGuidAlias(clientId, proxyUserId);
+        notifyModelUpdated(ModelChangeType.USER, proxyUserId);
+        return result;
+    }
+
+    @Override
+    public UserDao getUserDao() {
+        return target.getUserDao();
+    }
+
+    @Override
+    public UserGovernanceSql getUserGovernanceSql() {
+        return target.getUserGovernanceSql();
+    }
+
+    @Override
+    public int disableActiveProxies(long participantId, long studyId) {
+        List<Long> proxyIds = findGovernancesByParticipantAndStudyIds(participantId, studyId)
+                .map(governance -> governance.getProxyUserId())
+                .collect(Collectors.toList());
+        var result = target.disableActiveProxies(participantId, studyId);
+        proxyIds.forEach(proxyId -> notifyModelUpdated(ModelChangeType.USER, proxyId));
+        notifyModelUpdated(ModelChangeType.USER, participantId);
+        return result;
+    }
```

Comment:
This is an interesting case. This call actually affects governance but we can figure out only from its arguments who gets affected, so we need to do some additional work for that.